pkg/statedb/types.go

```diff
@@ -108,6 +108,20 @@ type Table[Obj ObjectConstraints[Obj]] interface {
 	// Writer when given a write transaction returns a table writer
 	// that can be used to modify the table.
 	Writer(tx WriteTransaction) TableReaderWriter[Obj]
+
+	// Initialized returns a channel that is closed when the table
+	// has been fully initialized. This method can only be used
+	// from start onwards.
+	Initialized() <-chan struct{}
+
+	// RegisterInitializer registers an initializer. Returns
+	// a channel that the initializer should close once it
+	// has completed.
+	//
+	// This method can only be called before starting in order
```

Comment:
Let's use the standard "MUST" etc terminology, so 
```suggestion
	// This method MUST be called before starting in order
```

also, what exactly is "starting" here? It's not like there's a `Start()` method on the interface?

Reply:
as in `hive.Start()`. e.g. it must be called from an constructor or invoke so that by the time we start all the registrations are gathered before we try and close the `initialized` channel. I'll try to improve the docs.